### PR TITLE
Add E2E tests for unqualified name call support on ODataRoute.

### PR DESCRIPTION
**Pull request by karataliu**
E2E tests for https://github.com/OData/WebApi/commit/68c56e30e5a1e839eec343d60d7b5f58088335c1

### CODE REVIEW

File: OData/test/E2ETest/WebStack.QA.Test.OData/Routing/UnqualifiedNameCallRoutingTests.cs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        public static void UpdateConfiguration(HttpConfiguration config)
+        {
+            var controllers = new[] { typeof(UnqualifiedCarsController) };
+            TestAssemblyResolver resolver = new TestAssemblyResolver(new TypesInjectionAssembly(controllers));
+
+            config.Services.Replace(typeof(IAssembliesResolver), resolver);
+            config.Routes.Clear();
+            config.EnableUnqualifiedNameCall(true);
+            config.MapODataServiceRoute("odata", "odata", GetModel());
+             config.EnsureInitialized();
```
Comment (xuzhg) [inline OData/test/E2ETest/WebStack.QA.Test.OData/Routing/UnqualifiedNameCallRoutingTests.cs]: align

File: OData/test/E2ETest/WebStack.QA.Test.OData/Routing/UnqualifiedNameCallRoutingTests.cs [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+            ODataModelBuilder builder = new ODataConventionModelBuilder();
+            EntitySetConfiguration<UnqualifiedCar> cars = builder.EntitySet<UnqualifiedCar>("UnqualifiedCars");
+            cars.EntityType.Action("Wash").Returns<string>();
+            cars.EntityType.Collection.Action("Wash").Returns<string>();
+            return builder.GetEdmModel();
+        }
+
+        [Theory]
+        [InlineData("/odata/UnqualifiedCars(5)/Wash", "WashSingle5")]
+        [InlineData("/odata/UnqualifiedCars/Wash", "WashCollection")]
```
Comment (xuzhg) [inline OData/test/E2ETest/WebStack.QA.Test.OData/Routing/UnqualifiedNameCallRoutingTests.cs]: add the namespace-qualified function call to verify the same result?

File: OData/test/E2ETest/WebStack.QA.Test.OData/Routing/UnqualifiedNameCallRoutingTests.cs [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        public void CanCallActionWithUnqualifiedRouteName(string url, string expectedResult)
```
Comment (xuzhg) [inline OData/test/E2ETest/WebStack.QA.Test.OData/Routing/UnqualifiedNameCallRoutingTests.cs]: CanCallFunction.... ?

File: OData/test/E2ETest/WebStack.QA.Test.OData/Routing/UnqualifiedNameCallRoutingTests.cs [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+            config.EnableUnqualifiedNameCall(true);
+            config.MapODataServiceRoute("odata", "odata", GetModel());
+             config.EnsureInitialized();
+        }
+
+        private static IEdmModel GetModel()
+        {
+            ODataModelBuilder builder = new ODataConventionModelBuilder();
+            EntitySetConfiguration<UnqualifiedCar> cars = builder.EntitySet<UnqualifiedCar>("UnqualifiedCars");
+            cars.EntityType.Action("Wash").Returns<string>();
```
Comment (xuzhg) [inline OData/test/E2ETest/WebStack.QA.Test.OData/Routing/UnqualifiedNameCallRoutingTests.cs]: add function?

File: OData/test/E2ETest/WebStack.QA.Test.OData/Routing/UnqualifiedNameCallRoutingTests.cs (hunk 1 above)
Comment (karataliu) [inline OData/test/E2ETest/WebStack.QA.Test.OData/Routing/UnqualifiedNameCallRoutingTests.cs]: verified that line not needed. removed.

File: OData/test/E2ETest/WebStack.QA.Test.OData/Routing/UnqualifiedNameCallRoutingTests.cs (hunk 2 above)
Comment (karataliu) [inline OData/test/E2ETest/WebStack.QA.Test.OData/Routing/UnqualifiedNameCallRoutingTests.cs]: added

File: OData/test/E2ETest/WebStack.QA.Test.OData/Routing/UnqualifiedNameCallRoutingTests.cs (hunk 4 above)
Comment (karataliu) [inline OData/test/E2ETest/WebStack.QA.Test.OData/Routing/UnqualifiedNameCallRoutingTests.cs]: added

**Comment by xuzhg** [comment]
Merged at:  3c2f90e

**Pull request closed without merging**